batch/batch/front_end/front_end.py

```diff
@@ -1075,6 +1077,95 @@ def check_service_account_permissions(user, sa):
     return await render_template('batch', request, userdata, 'job.html', page_context)
 
 
+async def _query_billing(request):
+    db = request.app['db']
+
+    date_format = '%m/%d/%Y'
+
+    default_start = datetime.datetime.now().replace(day=1)
+    default_start = datetime.datetime.strftime(default_start, date_format)
+
+    default_end = datetime.datetime.now()
+    default_end = datetime.datetime.strftime(default_end, date_format)
+
+    async def parse_error(msg):
+        session = await aiohttp_session.get_session(request)
+        set_message(session, msg, 'error')
+        return ([], default_start, default_end)
+
+    start_query = request.query.get('start', default_start)
+    try:
+        start = datetime.datetime.strptime(start_query, date_format)
+        start = start.timestamp() * 1000
+    except ValueError:
+        return await parse_error(f"Invalid value for start '{start_query}'; must be in the format of MM/DD/YYYY.")
+
+    end_query = request.query.get('end', default_end)
+    try:
+        end = datetime.datetime.strptime(end_query, date_format)
+        end = (end + datetime.timedelta(days=1)).timestamp() * 1000
+    except ValueError:
+        return await parse_error(f"Invalid value for end '{end_query}'; must be in the format of MM/DD/YYYY.")
+
+    if start > end:
+        return await parse_error(f'Invalid search; start must be earlier than end.')
+
+    sql = f'''
+SELECT billing_project, user, CAST(COALESCE(SUM(msec_mcpu), 0) AS SIGNED) AS msec_mcpu
+FROM batches
+WHERE batches.`time_created` >= %s AND batches.`time_created` <= %s
```

Comment:
This was the query you were worried about?  You've got the right index.  Looks good to me.  Ultimately we might want to limit the search range, but this seems great for now.